runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -202,6 +203,18 @@ std::unique_ptr<onert::ir::Model> loadModel(const std::string filename,
   return std::unique_ptr<onert::ir::Model>(nullptr);
 }
 
+std::unique_ptr<onert::ir::train::TrainingInfo>
+loadTrainingInfo(std::shared_ptr<onert::ir::Model> &&model)
```

Comment:
```suggestion
loadTrainingInfo(const std::shared_ptr<onert::ir::Model> &model)
```

Reply:
`extract_metadata` (used in line 212) is not a const function, It could not have `const` here.

Reply:
Also, `_nnpkg->primary_model()` (in this file, lie 292) returns shared_ptr; 
Here, shared_ptr itself looks like rvalue.

https://github.com/Samsung/ONE/blob/c925d8ae740700a1293d6a27f57c51e447650026/runtime/onert/core/include/ir/NNPkg.h#L92 

I'd like to reference shared_ptr itself (to not increase reference count), I used rvalue reference `std::shared_ptr<onert::ir::Model>&&`.

Reply:
> extract_metadata (used in line 212) is not a const function, It could not have const here.

`const std::shared_ptr<Model>` does not mean the object what the `std::shared_ptr` points to (`const Model *`), so it's OK if `const std::share` calls a const function of the object what it points to.

Reply:
> Also, _nnpkg->primary_model() (in this file, lie 292) returns shared_ptr.
Here, shared_ptr itself looks like rvalue.
> I'd like to reference shared_ptr itself (to not increase reference count), I used rvalue reference std::shared_ptr<onert::ir::Model>&&.

It seems to return value, not reference. I think there is no way not to increase the reference count unless you change `primary_model()`.

Reply:
Aha, Thank you a lot for clarifying - what really happened in this line :+1: :+1:

I updated it as you first suggested! - use const lvalue reference. :smile: